pithos/pithos.py

```diff
@@ -1460,22 +1463,63 @@ def restore_position(self):
         if not x is None and not y is None:
             self.move(int(x), int(y))
 
+    def restore_size(self):
+        """ resizes window to size stored in preferences """
+        w, h = self.settings['win-size']
+        if w is not None and h is not None:
+            self.resize(int(w), int(h))
+
+    def restore_maximized(self):
+        if self.settings['win-maxed']:
+            self.maximize()
+
+    def restore_window(self):
+        # Getting the window position
+        # does not work in Wayland.
+        if self.running_in_x:
+            self.restore_position()
+        self.restore_size()
+        self.restore_maximized()
+
     def bring_to_top(self, *ignore):
         timestamp = Gtk.get_current_event_time()
         self.present_with_time(timestamp)
 
     def present_with_time(self, timestamp):
-        self.restore_position()
+        self.restore_window()
         Gtk.Window.present_with_time(self, timestamp)
 
     def present(self):
-        self.restore_position()
+        self.restore_window()
         Gtk.Window.present(self)
 
     @GtkTemplate.Callback
-    def on_configure_event(self, widget, event, data=None):
-        x, y = self.get_position()
-        self.settings.set_value('win-pos', GLib.Variant('(ii)', (x, y)))
+    def on_configure_event(self, widget, *ignore):
+        def store_window_size_and_position():
+            self.configure_event_timer_id = 0
+            # Getting the window position
+            # does not work in Wayland.
+            if self.running_in_x:
+                x, y = widget.get_position()
+                self.settings.set_value('win-pos', GLib.Variant('(ii)', (x, y)))
+            w, h = widget.get_size()
+            self.settings.set_value('win-size', GLib.Variant('(ii)', (w, h)))
+            return False
+        # Debouncer. Don't hammer settings.
+        # Store the settings after the configure
+        # events stop comming.
+        if self.configure_event_timer_id:
```

Comment:
So I honestly question if this actually improves anything. Making new GSources is a relatively expensive operation as I understand it so you replaced writing to DConf repeatedly which is a queued write operation and is fast with a slower global glib lock to a queued operation.

Now I might be wrong but without profiling I don't think we should make our code much uglier unless its provably a performance win.

Reply:
Another reason for this is that it gives the window time to actually finish resizing before we check the size. If you go straight from a configure event to checking the size you can end up with a stale size.